### ISSUE TITLE
Add the capability of setting a tolerance level on validation for early stopping in LightGBMRanker training [Spark/scala]

### ISSUE THREAD

**Issue by catenamatteo** [Azure/mmlspark#713]
Even by using a validation set and enabling early stopping, the training of LightGBMRanker often reaches the maximum number of iterations. However, the metric computed using the validation set often differs by only a small margin between two iterations, something after the 4 decimal point. 

Wouldn't be convenient to be able to set a tolerance t, such as that training stops after x rounds if the evaluation metric has improved less than t in x iterations? This should speed up the training phase, no?

**Comment by JoanFM** [comment]
PR https://github.com/Azure/mmlspark/pull/786 should be enough to provide this requirement

**Comment by imatiach-msft** [comment]
closing issue as this has been fixed and released, thank you!  Please reopen if I misunderstood and there is still an issue.
